src/types/ListOptions.ts

```diff
@@ -1,10 +1,7 @@
 export default interface ListOptions {
-    beforeId?: string;
-    beforeValue?: string;
-    afterId?: string;
-    afterValue?: string;
+    prevCursor?: string;
```

Comment:
It might be worth renaming this file to `List.ts` and adding a common `ListResponse` type:
```typescript
export interface ListResponse<T> {
   results: []T;
   prevCursor?: string;
   nextCursor?: string;
}
```

Then you can just make use of the generic type on all list methods:
```typescript
public static async query(query: string, listOptions: ListOptions = {}, options: WarrantRequestOptions = {}): Promise<ListResponse<QueryResult>> {
   // ...
}

public async listFeatures(listOptions: ListFeatureOptions = {}, options: WarrantRequestOptions = {}): Promise<ListResponse<Feature>> {
   // ...
}
```